internal/vds/vds.go

```diff
@@ -503,24 +512,56 @@ func (v VDSHandle) GetAttributes(
 	}
 
 	defer C.response_delete(&horizon)
-	
+
 	buffer := make([]byte, mapsize * len(targetAttributes))
-	cerr = C.attribute(
-		v.context(),
-		v.Handle(),
-		surface.get(),
-		horizon.data,
-		horizon.size,
-		&cattributes[0],
-		C.size_t(len(targetAttributes)),
-		C.float(above),
-		C.float(below),
-		C.float(stepsize),
-		unsafe.Pointer(&buffer[0]),
-	)
 
-	if err := v.Error(cerr); err != nil {
-		return nil, err
+	maxConcurrency := 32
+	windowsPerRoutine := int(math.Ceil(float64(hsize) / float64(maxConcurrency)))
```

Comment:
Question: do you think there would be a performance problem anywhere here with jumping logic?
If hsize = 96, we have 32 threads with each window size 3.
If hsize = 97, suddenly we have 25 threads - 24 with window size 4 and one with window size 1.

I am wondering if it would be worse, better or the same to keep number of threads consistent?
Like if hsize = 97, have 32 threads - 31 with window size 3 and 1 with window size 4.

Second approach probably complicates logic, but I can't really tell if it would matter.

Reply:
From what I've tested it's definitly not worth it. The current worst case is that the last routine only gets 1 window. I.e. it will finish right away an in practise we are utilising one less go routine. On my machine it doesn't seam to do much if I double or half the numbers of goroutines, so I don't think this will matter much